### PR TITLE
Implemented `LogicNetwork.read`. 

### PR DESCRIPTION
**Pull request by siyuzhou**
Added myeloid logic table data in `data` folder. Added a few test cases for the new function and example data.

### CODE REVIEW

File: neet/boolean/examples.py [hunk 1]
```diff
@@ -58,65 +58,70 @@
 # Get the path of the nodes and edges files for the mouse cortical gene regulatory network
 # Edges are from either figure 7B or 7C (Giacomantonio, 2010)
 
-MOUSE_CORTICAL_ANT_INIT_NODES = join(DATA_PATH, "mouse_cortical_anterior_init-nodes.txt")
-MOUSE_CORTICAL_ANT_FINAL_NODES = join(DATA_PATH, "mouse_cortical_anterior_final-nodes.txt")
-MOUSE_CORTICAL_POST_INIT_NODES = join(DATA_PATH, "mouse_cortical_posterior_init-nodes.txt")
-MOUSE_CORTICAL_POST_FINAL_NODES = join(DATA_PATH, "mouse_cortical_posterior_final-nodes.txt")
+MOUSE_CORTICAL_ANT_INIT_NODES = join(
+    DATA_PATH, "mouse_cortical_anterior_init-nodes.txt")
```
Comment (hbsmith) [inline neet/boolean/examples.py]: are these intentionally formatted to be on two lines?

File: neet/boolean/logicnetwork.py [hunk 2]
```diff
@@ -48,35 +48,46 @@ def __init__(self, table):
 
         ::
 
-            >>> net = LogicNetwork([((1, 2), {'01', '10'}),
+            >>> net = LogicNetwork([((1, 2), {'01', '10'}, ['A', 'B']),
```
Comment (hbsmith) [inline neet/boolean/logicnetwork.py]: Does this actually work? Or should the `['A', 'B']` be after the table list?

File: neet/boolean/logicnetwork.py [hunk 3]
```diff
@@ -210,20 +220,128 @@ def update(self, net_state, index=None, pin=None):
             [0, 0, 0]
         """
         if net_state not in self.state_space:
-            raise ValueError("the provided state is not in the network's state space")
+            raise ValueError(
+                "the provided state is not in the network's state space")
```
Comment (hbsmith) [inline neet/boolean/logicnetwork.py]: Is this meant to be on two lines as well?

File: neet/boolean/logicnetwork.py (hunk 2 above)
Comment (siyuzhou) [inline neet/boolean/logicnetwork.py]: Good catch.

File: neet/boolean/logicnetwork.py (hunk 3 above)
Comment (siyuzhou) [inline neet/boolean/logicnetwork.py]: My auto-format plugin reformats any line longer than 79 characters into multiple lines according to PEP8.

**Pull request merged**